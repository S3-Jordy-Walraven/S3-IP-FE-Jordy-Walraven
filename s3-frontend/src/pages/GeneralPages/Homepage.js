import React from "react";
import EffectTableComponent from "../../components/EffectComponents/Table/EffectTableComponent";


const Homepage = (props) => {
  return (
    <div data-testid="homepage-1">
      <div>
        <h1 style={{ textAlign: "center", marginTop: "20px", color: "white" }}>SIGNIFY</h1>
      </div>
      <EffectTableComponent effects={props.allEffects} />
    </div>
  );
};

export default Homepage;
