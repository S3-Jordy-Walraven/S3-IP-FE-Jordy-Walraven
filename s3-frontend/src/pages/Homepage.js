import React, { useEffect } from "react";
import AccountService from "../services/AccountService";
const Homepage = () => {
  const service = new AccountService();
  useEffect(() => {
   service.loginUser();
  });

  return (
    <div>
      <h1>HomePage</h1>
    </div>
  );
};

export default Homepage;
