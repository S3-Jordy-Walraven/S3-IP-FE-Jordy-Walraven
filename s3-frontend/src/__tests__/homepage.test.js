import { render, screen, cleanup } from '@testing-library/react';
import renderer from 'react-test-renderer';
import Homepage from "../pages/Homepage"


const dummyEffects = { data: [] }
dummyEffects.data = [
    {
        "id": 1,
        "effectName": "Effect 1",
        "effectContent": "<script>console.log('hello')</script>",
        "creatorName": "Creator 1",
    },
    {
        "id": 2,
        "effectName": "Effect 2",
        "effectContent": "<script>console.log('hello')</script>",
        "creatorName": "Creator 2",
    },
    {
        "id": 3,
        "effectName": "Effect 3",
        "effectContent": "<script>console.log('hello')</script>",
        "creatorName": "Creator 3",
    }
]

afterEach(cleanup)

test('Should render 3 effects', () => {
    render(<Homepage allEffects={dummyEffects} />);
    const homePageElement = screen.getByTestId("homepage-1");
    expect(homePageElement).toBeInTheDocument();
    expect(homePageElement).toHaveTextContent("Signify");
    expect(homePageElement).toHaveTextContent("Effect 1");
    expect(homePageElement).toHaveTextContent("Effect 2");
    expect(homePageElement).toHaveTextContent("Creator 3");
    expect(homePageElement).toContainHTML("<script>console.log('hello')</script>");
})

test('Should render 0 effects', () => {
    render(<Homepage allEffects={{ data: [] }} />);
    const homePageElement = screen.getByTestId("homepage-1");
    expect(homePageElement).toBeInTheDocument();
    expect(homePageElement).toHaveTextContent("Signify");
    expect(homePageElement).not.toHaveTextContent("Effect 1");
    expect(homePageElement).not.toHaveTextContent("Effect 2");
    expect(homePageElement).not.toHaveTextContent("Creator 3");
    expect(homePageElement).not.toContainHTML("<script>console.log('hello')</script>");
})

test("matches snapshot", () => {
    const tree = renderer.create(<Homepage allEffects={dummyEffects} />).toJSON();
    expect(tree).toMatchSnapshot();
})