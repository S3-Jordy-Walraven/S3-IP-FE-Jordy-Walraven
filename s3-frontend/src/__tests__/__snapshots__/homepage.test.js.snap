// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  data-testid="homepage-1"
>
  <div>
    <h1
      style={
        Object {
          "color": "white",
          "marginTop": "20px",
          "textAlign": "center",
        }
      }
    >
      SIGNIFY
    </h1>
  </div>
  <div
    className="container-fluid"
  >
    <div
      className="row justify-content-md-center"
    />
  </div>
</div>
`;
